src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsUpdaterResult.cs

```diff
@@ -5,7 +5,13 @@
 
 namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
 {
-    public record WinformsUpdaterResult(bool Result, string Message, IList<string> FileLocations) : IUpdaterResult
+    public record WinformsUpdaterResult(
+        string RuleId,
+        string RuleName,
+        string FullDescription,
+        bool Result,
+        string Message,
+        IList<string> FileLocations) : IUpdaterResult
```

Comment:
Are there cases where we don't have a message or FileLocations?
I was thinking in adding everything to the IUpdaterResult and just use the interface properties instead of casting to WindowsUpdaterResult in the WindowsDefaultFontUpdater, for example

Reply:
Yeah there are multiple source updaters under Windows extensions and I think they all use the same UpdaterResult and can have FileLocations empty.

Reply:
Does it have to be a list? If so, I think I'd prefer an `IReadOnlyList<>` or `ImmutableArray` (something that cannot be changed)